src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/azuredevops/AzureDevOpsServerPullRequestDecorator.java

```diff
@@ -0,0 +1,407 @@
+                        LOGGER.trace(String.format("ISSUE: getRuleKey: %s ", issue.getIssue().getRuleKey()));
+                        LOGGER.trace(String.format("COMPONENT: getDescription: %s ", issue.getComponent().getDescription()));
+                        DbIssues.Locations locate = Objects.requireNonNull(issue.getIssue().getLocations());
+                        boolean isExitsThread = false;
+                        for (CommentThread azureThread : azureCommentThreads) {
+                            LOGGER.trace(String.format("azureFilePath: %s", azureThread.getThreadContext().getFilePath()));
+                            LOGGER.trace(String.format("filePath: %s (%s)", filePath, azureThread.getThreadContext().getFilePath().equals(filePath)));
+                            LOGGER.trace(String.format("azureLine: %d", azureThread.getThreadContext().getRightFileStart().getLine()));
+                            LOGGER.trace(String.format("line: %d (%s)", line, azureThread.getThreadContext().getRightFileStart().getLine() == locate.getTextRange().getEndLine()));
+
+                            if (azureThread.getThreadContext().getFilePath().equals(filePath)
+                                    && azureThread.getComments()
+                                    .stream()
+                                    .filter(c -> c.getContent().contains(issue.getIssue().key()))
+                                    .count() > 0 ) {
+
+                                if(issue.getIssue().getStatus() != Issue.STATUS_OPEN
+                                        && azureThread.getStatus() == CommentThreadStatus.ACTIVE) {
+                                    Comment comment = new Comment("Closed in SonarQube");
+                                    LOGGER.info("Issue closed in Sonar. try close in Azure");
+                                    sendPost(
+                                            azureThread.getLinks().getSelf().getHref() + "/comments" + getApiVersion(apiVersion),
+                                            new ObjectMapper().writeValueAsString(comment),
+                                            "Comment added success",
+                                            authorizationHeader
+                                    );
+                                    sendPatch(
+                                            azureThread.getLinks().getSelf().getHref() + getApiVersion(apiVersion),
+                                            "{\"status\":\"closed\"}",
+                                            authorizationHeader
+                                    );
+                                }
+                                isExitsThread = true;
+                                break;
+                            }
+                        }
+                        if (!issue.getIssue().getStatus().equals(Issue.STATUS_OPEN)) {
```

Comment:
From previous PR (#202 ) - Use == and != for enum comparison

Reply:
Fixed

Reply:
I'm not a java dev, so I have only copied that comment from previous PR, but now I can see that sonar marks the comparison using == as error. I think, that you should use equals, because status is not enum but string - http://javadocs.sonarsource.org/5.4/apidocs/org/sonar/api/issue/Issue.html#status()
Sorry.